src/lib.rs

```diff
@@ -92,12 +159,12 @@ mod tests {
 
     #[test]
     fn valid_username_check() {
-        assert!(is_valid_username("a"));
-        assert!(is_valid_username("Test_test tHAt-tester."));
-        assert!(is_valid_username("1234567890"));
-        assert!(is_valid_username("l33t 5p34k"));
-        assert!(!is_valid_username(""));
-        assert!(!is_valid_username("Test! @thing"));
-        assert!(!is_valid_username("123456789111315171921232527293133"));
+        assert!(is_valid_username("a").is_ok());
+        assert!(is_valid_username("Test_test tHAt-tester.").is_ok());
+        assert!(is_valid_username("1234567890").is_ok());
+        assert!(is_valid_username("l33t 5p34k").is_ok());
+        assert!(is_valid_username("").is_err());
+        assert!(is_valid_username("Test! @thing").is_err());
+        assert!(is_valid_username("123456789111315171921232527293133").is_err());
```

Comment:
I think you should keep the original `is_valid_username` but add a new function with name `check_valid_username`.